### ISSUE TITLE
Reverse DNS Lookups

### ISSUE THREAD

**Issue by rweckhoff** [nitaybz/homebridge-network-presence#4]
First of all ... nice work.

I do not know if this is a problem or not, however, I am noticing a significant amount of reverse DNS lookups occurring while this plugin is installed and enabled.  Polling for 4 devices (Mac address or IP) is generating thousands of DNS queries every 5 minutes.

**Comment by gdotp01** [comment]
Yep,I’m seeing the same. I noticed it tonight when checking my pi.hole server.

**Comment by nitaybz** [comment]
I released a new version that sets the default polling to "every 10 seconds" instead of 1 and the ability to change the interval to the desired amount of time in seconds.
hope it reduces the traffic to an acceptable value...

**Comment by nitaybz** [comment]
https://github.com/nitaybz/homebridge-network-presence/releases/tag/v1.0.6

**Comment by nitaybz** [comment]
Please try v1.2.0...
I've optimize the network scanner, please test it

**Comment by rweckhoff** [comment]
Upgraded to 1.2.1

Same observation. 

Screenshot at 4:02 / with DNS exclusions for local .arpa requests 

Screenshot at 4:07 / without DNS exclusions. Almost 8K in DNS lookups. 30 second polling configured. 

Screenshot at 4:16 / with DNS exclusions for local .arpa requests back in place

![image](https://user-images.githubusercontent.com/61269490/103211005-53b39500-48c4-11eb-85a3-53370f71f981.png)

![image](https://user-images.githubusercontent.com/61269490/103211025-5910df80-48c4-11eb-9243-76ebf9d20818.png)

![image](https://user-images.githubusercontent.com/61269490/103211037-5e6e2a00-48c4-11eb-9f18-bea19a9e5541.png)

**Comment by nitaybz** [comment]
"8K in DNS lookups." - is it in 5 minutes? means 800 on each request?
is it only the plugin that creates those lookups? please share more specific information if you can...

**Comment by nitaybz** [comment]
is it possible that you have multiple subnets for this plugin to look into? do you have a normal network setup?
will it help if i'll add ip range or subnet range so it will not look through your entire network?

**Comment by rweckhoff** [comment]
Sorry away from the computer, however, here is the screenshot of my network.  Homebridge and AdGuard running on docker containers. 192.168.1.x is my primary network. Second screenshot shows reverse lookup traffic on the 192.168 and 10.0 network ranges. 





Thanks,

Bob

**Comment by rweckhoff** [comment]
Sorry away from the computer, however, here is the screenshot of my network. Homebridge and AdGuard running on docker containers. 192.168.1.x is my primary network. Second screenshot shows reverse lookup traffic on the 192.168 and 10.0 network ranges. 

Thanks, Bob

![image](https://user-images.githubusercontent.com/61269490/103228639-7c508480-48ee-11eb-9217-f20d57e03edd.png)

![image](https://user-images.githubusercontent.com/61269490/103228626-778bd080-48ee-11eb-808d-897a08ab87f4.png)

**Comment by nitaybz** [comment]
So I think the amount of traffic is due to the complexity of your network... I'll add an option to set IP range or subnet range and it should help you reduce the amount of traffic by focusing the scan on the desired range.

**Comment by jealousofyou** [comment]
I've used the addressRange option in v1.2.3 and Pi-Hole shows that the Homebridge device is still doing lookups of IP addresses outside that range.

**Comment by julianq** [comment]
Realize this is a bit of thread necromancy, but I'm having the same issue. I have my PiHole to refuse requests to a client after 1,000 DNS requests within 60 seconds, and this plugin is hitting that limit constantly.

I've increased polling interval and set the address range.

**Comment by nitaybz** [comment]
Closing this. I'm not actively using Homebridge anymore and won't be adding new features to this plugin. If anyone wants to send a clean PR I am happy to review and release.

Sorry for the long delay on this.
